### DISCUSSION TITLE
Require linear history except *some branch*

### DISCUSSION THREAD

**Discussion by jmvtrinidad** [community/community#41794] [Repositories]
### Select Topic Area

Product Feedback

### Body

Is it possible to have an option to have some branch excluded for linear commit?
every other branch can be squash and merge except some branch.
For example having a workflow of main(development) and staging branch (preprod/prod), we normally merge commit for synching between this two branch and not allowing to squash and merge for synching this 2 branch. but merging to this branch with feature branch we normally do squash and merge.
If we have this option, it will enforce that rule.
![image](https://user-images.githubusercontent.com/7194300/207771826-9ed16bd7-cc2b-4718-8692-3a36434a7913.png)

There is similar discussion like this [one](https://github.com/community/community/discussions/10809).
But it differs in a way that we want linear history (squash merge) for main and staging for all branch except main to staging or staging to main (merge commit). And on the other discussion they want to exclude squash and merge for a specific branch only allow merge commit.

**Comment by jfullstackdev** [comment]
just want to share mine

my personal workflow that I'm using in my company is to use branch, if every developer will do the same, it will lessen the merge conflict

## Steps ( GitHub Desktop )
1. Always fetch and pull it.

2. Start coding.

3. When it's time to commit it, don't commit it on the default branch. Make a branch instead and bring the changes to it, so there will be no new commits left on the default branch.

4. My way of naming new branches is the date and the count of the commit, like `december12-sprint1`.

5. I can get the `commit summary` by  using the `tab` key but if I need to make a very important summary I will manually do so.

6. Once it is committed, I will publish the branch. This is an unverified commit on GitHub. Then the branch will pop up on GitHub. I'll review it and make a PR then merge. This is a verified commit now. This is clean actually. When it's not working at the production setup, you can always revert it on GitHub.

7. I will switch to master leaving the changes on that branch. Then I will fetch again and pull those recent changes. Done. You have the synced local project from GitHub. Start working on code again.
